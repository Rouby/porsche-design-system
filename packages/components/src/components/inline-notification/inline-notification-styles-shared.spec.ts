import {
  getNotificationContentJssStyle,
  getNotificationIconJssStyle,
  getNotificationRootJssStyle,
} from './inline-notification-styles-shared';

describe('getNotificationRootJssStyles()', () => {
  it.each<Parameters<typeof getNotificationRootJssStyle>>([
    ['info', false, false, 'light'],
    ['info', true, false, 'light'],
    ['info', true, true, 'light'],
    ['info', false, true, 'light'],
    ['neutral', false, false, 'light'],
    ['success', false, false, 'light'],
    ['error', false, false, 'light'],
    ['warning', false, false, 'light'],
    ['info', false, false, 'dark'],
    ['neutral', false, false, 'dark'],
    ['success', false, false, 'dark'],
    ['error', false, false, 'dark'],
    ['warning', false, false, 'dark'],
  ])('should return correct JssStyle for state: %s and theme: %s', (...args) => {
    expect(getNotificationRootJssStyle(...args)).toMatchSnapshot();
  });
});

describe('getNotificationIconJssStyle()', () => {
  it('should return correct JssStyle', () => {
    expect(getNotificationIconJssStyle()).toMatchSnapshot();
  });
});

describe('getNotificationContentJssStyle()', () => {
  it('should return correct JssStyle', () => {
    expect(getNotificationContentJssStyle()).toMatchSnapshot();
  });
});
