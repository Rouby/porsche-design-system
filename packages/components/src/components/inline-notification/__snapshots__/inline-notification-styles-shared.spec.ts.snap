// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNotificationContentJssStyle() should return correct JssStyle 1`] = `
{
  "@media(min-width:760px)": {
    "marginLeft": "-8px",
  },
  "display": "grid",
  "gap": "4px",
  "maxWidth": "50rem",
}
`;

exports[`getNotificationIconJssStyle() should return correct JssStyle 1`] = `
{
  "@media(max-width:759px)": {
    "display": "none",
  },
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: error and theme: false 1`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#FCE8E9",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: error and theme: false 2`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#290A0A",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: info and theme: false 1`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#E8EEFD",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: info and theme: false 2`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr) auto",
  },
  "background": "#E8EEFD",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr) auto",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: info and theme: false 3`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#001A33",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: info and theme: true 1`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr) auto",
  },
  "background": "#E8EEFD",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: info and theme: true 2`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr) auto auto",
  },
  "background": "#E8EEFD",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr) auto",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: neutral and theme: false 1`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#E8EEFD",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: neutral and theme: false 2`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#001A33",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: success and theme: false 1`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#EBFAF0",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: success and theme: false 2`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#003320",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: warning and theme: false 1`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#FFF9E6",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;

exports[`getNotificationRootJssStyles() should return correct JssStyle for state: warning and theme: false 2`] = `
{
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto minmax(auto, 1fr)",
  },
  "background": "#2B2208",
  "borderRadius": "4px",
  "display": "grid",
  "gap": "16px",
  "gridTemplateColumns": "minmax(auto, 1fr)",
  "padding": "16px",
  "placeItems": "start",
}
`;
